libevm/VMOpt.cpp

```diff
@@ -49,20 +49,22 @@ void VM::initMetrics()
 	done = true;
 }
 
-//
-// init interpreter on entry
-//
-
+/// Init interpreter on entry.
 void VM::initEntry()
 {
-	m_bounce = &VM::interpretCases;	
+	m_bounce = &VM::interpretCases;
+
+	// Copy and extend code by 32 bytes to handle virtual push data.
+	auto extendedSize = m_ext->code.size() + 32;
+	m_codeSpace.reserve(extendedSize);
 	m_codeSpace = m_ext->code;
+	m_codeSpace.resize(extendedSize);
```

Comment:
Alternatively, we can but put back code to pad on the fly
https://github.com/ethereum/cpp-ethereum/blob/push-beyond-limits/libevm/VM.cpp#L774